File: leetcode/1.twoSum/twoSum.go
package twoSum

func twoSum(nums []int, target int) []int {
	numHash := make(map[int]int)
	for index, value := range nums {
		numHash[value] = index
	}
	for index, value := range nums {
		tmpVal := target - value
		if v, ok := numHash[tmpVal]; ok {
			if index != v {
				return []int{index, v}
			}
		}
	}

	return nil
}

File: Algorithms/src/main/java/basic/TwoSum.java
package basic;
import java.util.HashMap;
import java.util.ArrayList;
import java.util.List;


public class TwoSum {

    public static List<Integer> twoSum(List<Integer> nums, Integer target) {
        HashMap<Integer, Integer> numHash = new HashMap<Integer, Integer>();
        List<Integer> l = new ArrayList<Integer>();
        for (int i = 0; i< nums.size(); i++) {
            numHash.put(nums.get(i), i);
        }
        for (int i = 0; i<nums.size(); i++) {
            int tmpVal = target - nums.get(i);
            Integer value = numHash.get(tmpVal);
            if ( value != null && i != value ) {
                l.add(i);
                l.add(value);
                break;
            }
        }
        return l;
    }
}
